python/pyspark/sql/connect/plan.py

```diff
@@ -143,23 +143,29 @@ def __init__(self, child: Optional["LogicalPlan"], *columns: "ExpressionOrString
         self._verify_expressions()
 
     def _verify_expressions(self) -> None:
-        """Ensures that all input arguments are instances of Expression."""
+        """Ensures that all input arguments are instances of Expression or String."""
         for c in self._raw_columns:
-            if not isinstance(c, Expression):
-                raise InputValidationError(f"Only Expressions can be used for projections: '{c}'.")
+            if not isinstance(c, (Expression, str)):
+                raise InputValidationError(
+                    f"Only Expressions or String can be used for projections: '{c}'."
+                )
 
     def withAlias(self, alias: str) -> LogicalPlan:
         self.alias = alias
         return self
 
     def plan(self, session: Optional["RemoteSparkSession"]) -> proto.Relation:
         assert self._child is not None
-        proj_exprs = [
-            c.to_plan(session)
-            if isinstance(c, Expression)
-            else self.unresolved_attr(*(c.split(".")))
-            for c in self._raw_columns
-        ]
+        proj_exprs = []
+        for c in self._raw_columns:
+            if isinstance(c, Expression):
+                proj_exprs.append(c.to_plan(session))
+            elif c == "*":
+                exp = proto.Expression()
+                exp.unresolved_star.SetInParent()
+                proj_exprs.append(exp)
+            else:
+                proj_exprs.append(self.unresolved_attr(*(c.split("."))))
```

Comment:
Hm, I feel like we shouldn't/can't handle the dot separation here. What about backquotes? And it would be more compliated when we support other `GetItem-like` syntax here

Reply:
First of all this belongs to 

In fact you asking reminds me a very tricky problem that I didn't think of: in the past PySpark does not care because it can use Py4j to reuse Scala implementation, however now the world has changed....

For this specific problem, it is not about the separator. What we really need to do is to patch the scala implementation for `CatalystSqlParser.parseMultipartIdentifier(s)`. The identifier could contain multiple parts.

Reply:
It is tricky. How about let me talk to Wenchen to confirm `CatalystSqlParser.parseMultipartIdentifier(s)`'s behavior? I can follow up as a separate task for how to parse multiple name parts correctly on python side?

Reply:
I hope there are only a few such cases, otherwise we might have a bit more risks than expected....

Reply:
I am fine with going ahead for now but WDYT @cloud-fan ?

Reply:
After some thoughts, I think there is another way which is to not ask clients to parse the identifier so clients only need to provide the relation name. I put my idea into this draft PR: https://github.com/apache/spark/pull/38264

Reply:
I can close this conversion now. The solution (https://github.com/apache/spark/pull/38264) was already merged.